pkg/commands/cmdlib/bundleio.go

```diff
@@ -78,15 +119,32 @@ func inlineData(ctx context.Context, data *bundle.Bundle, rw files.FileReaderWri
 	}
 	var err error
 	if g.InlineComponents {
-		data, err = inliner.InlineBundleFiles(ctx, data)
-		if err != nil {
-			return nil, fmt.Errorf("error inlining component data files: %v", err)
+		switch v := data.(type) {
+		case *bundle.Bundle:
+			data, err = inliner.InlineBundleFiles(ctx, v)
+			if err != nil {
+				return nil, fmt.Errorf("error inlining component data files: %v", err)
+			}
+		case *bundle.ComponentPackage:
+			data, err = inliner.InlineComponent(ctx, v)
```

Comment:
maybe this should be moved to InlineObjects?. InlineComponent is perhaps little misnamed. It inlines the objects in the components.